File: frontend/src/app/core/models/dto/charging-location-detail.dto.ts
import { ChargingStationSummaryDto } from './charging-station-summary.dto';

export interface ChargingLocationDetailDto {
  id: number;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  imagePath: string | null;
  stations: ChargingStationSummaryDto[];
  stationsCount: number;
}

File: backend/src/main/java/com/powerme/dto/ChargingLocationDetailDto.java
package com.powerme.dto;

import java.util.List;

public record ChargingLocationDetailDto(
        Long id,
        String name,
        String address,
        double latitude,
        double longitude,
        String imagePath,
        List<ChargingStationDto> stations,
        int stationsCount
) {
    // Factory
    public static ChargingLocationDetailDto of(
            ChargingLocationDetailDto base,
            List<ChargingStationDto> stations
    ) {
        return new ChargingLocationDetailDto(
                base.id(),
                base.name(),
                base.address(),
                base.latitude(),
                base.longitude(),
                base.imagePath,
                stations,
                stations.size()
        );
    }
}
